src/mimeparser.rs

```diff
@@ -1295,22 +1254,12 @@ impl MimeMessage {
             }
         }
         peerstate.public_key = Some(key);
-        if mail_is_encrypted {
-            info!(
-                context,
-                "using attached PGP key for peer '{}' with prefer-encrypt=mutual as the mail is \
-                encrypted",
-                peerstate.addr,
-            );
-            peerstate.prefer_encrypt = EncryptPreference::Mutual;
-            peerstate.save_to_db(&context.sql).await?;
-        } else {
-            info!(
-                context,
-                "using attached PGP key for peer '{}'", peerstate.addr,
-            );
-            peerstate.prefer_encrypt = EncryptPreference::NoPreference;
```

Comment:
I think it's not worth the code complexity. Anyway for encrypted messages we set `EncryptionPreference::Mutual`, so let's do that for all